api/controller.py

```diff
@@ -478,8 +485,10 @@ def search(self, languages, lane_name):
         if isinstance(lane, ProblemDetail):
             return lane
         query = flask.request.args.get('q')
+        library_short_name = flask.request.library.short_name
         this_url = self.url_for(
             'lane_search', languages=languages, lane_name=lane_name,
+            library_short_name=library_short_name,
```

Comment:
I feel like there ought to be a way to make self.url_for automatically fill in `library_short_name=flask.request.library.short_name` if it's not provided. Most of the time, the library in the link will be the same library used in the request.